File: app/src/test/java/com/readtracker/android/db/QuoteTest.java
package com.readtracker.android.db;

import android.test.AndroidTestCase;

import org.junit.Test;

public class QuoteTest extends AndroidTestCase {

  /**
   * Create a book entity and set a quote on it. Merge this information
   * into another book entity and assert the information.
   */
  @Test
  public void quoteTest_MergeInformationFromBookToBook_ReturnsBookCopy() {
    Book book = new Book();
    Quote original = new Quote();

    original.setContent("I have content");
    original.setAddTimestampMs(123456789L);
    original.setPosition(0.45f);
    original.setBook(book);

    Quote merged = new Quote();
    merged.merge(original);

    assertEquals("I have content", merged.getContent());
    assertEquals(Long.valueOf(123456789L), merged.getAddTimestampMs());
    assertEquals(0.45f, merged.getPosition());
    assertSame(book, merged.getBook());
  }
}

File: app/src/test/java/com/readtracker/android/db/QuoteTest.kt
package com.readtracker.android.db

import org.junit.Assert.assertEquals
import org.junit.Assert.assertSame
import org.junit.Test

class QuoteTest {

    /**
     * Create a [Book] entity and set a [Quote] on it. Merge this information
     * into another book entity and assert the information.
     */
    @Test
    fun quoteTest_MergeInformationFromBookToBook_ReturnsBookCopy() {
        val book = Book()

        val original = Quote().apply {
            content = "I have content"
            addTimestampMs = 123456789L
            position = 0.45f
            this@apply.book = book
        }

        val merged = Quote().apply {
            merge(original)
        }

        assertEquals("I have content", merged.content)
        assertEquals(123456789L, merged.addTimestampMs)
        assertEquals(0.45f, merged.position)
        assertSame(book, merged.book)
    }
}